tests/unit/test_expression_tree/test_binary_operators.py

```diff
@@ -668,6 +668,24 @@ def test_inner_simplifications(self):
         self.assertEqual(pybamm.inner(a3, a2).evaluate(), 3)
         self.assertEqual(pybamm.inner(a3, a3).evaluate(), 9)
 
+    def test_to_equation(self):
+        # Test Power
+        self.assertEqual(pybamm.Power(7, 2).to_equation(), 49)
+
+        # Test Division
+        self.assertEqual(pybamm.Division(10, 2).to_equation(), 5)
+
+        # Test Matrix Multiplication
+        arr1 = pybamm.Array([[1, 0], [0, 1]])
+        arr2 = pybamm.Array([[4, 1], [2, 2]])
+        self.assertEqual(
+            pybamm.MatrixMultiplication(arr1, arr2).to_equation(),
+            sympy.Matrix([[4.0, 1.0], [2.0, 2.0]]),
+        )
+
+        # Test Heaviside
+        self.assertEqual(pybamm.Heaviside("test", 4, 2).to_equation(), 1)
```

Comment:
The same code works on the jupyter notebook and gives `1` but here gives an error

Reply:
It works in notebook but not in test because these lines are only called in the tests when debug mode is true https://github.com/pybamm-team/PyBaMM/blob/08de41714aa093fa2b705a649d4132d3418d43b3/pybamm/expression_tree/symbol.py#L226-L232

Reply:
`_binary_evaluate` isn't implemented for `Heaviside`. Instead of calling `Heaviside` directly just test both `EqualHeaviside` and `NotEqualHeaviside`. Can you also change the class name `Heaviside` to `_Heaviside` to indicate that it is a semi-private class (shouldn't be called directly)?

Reply:
Thanks, @tinosulzer, I have removed the `to_equation` from the Heaviside as it is a semi-private class and won't be called directly

Reply:
`to_equation` can still be in the Heaviside if the behavior of EqualHeaviside and NotEqualHeavside is intented to be the same (saves writing the same code twice). Then `EqualHeaviside.to_equation` will just call `_Heaviside.to_equation`

Reply:
The `_binary_evaluate` in EqualHeaviside and NotEqualHeavside already does what `sympy.Heaviside` is intended to do, so I think we can probably skip overriding the `to_equation` for the _Heaviside class